app/stores/DocumentsStore.ts

```diff
@@ -486,9 +498,16 @@ export default class DocumentsStore extends BaseStore<Document> {
       const document = this.data.get(res.data.document.id);
       invariant(document, "Document not available");
 
+      if (options.shareId) {
```

Comment:
This worked out quite nicely, we could probably combine `fetch` and `fetchWithSharedTree` methods now if we wanted. Perhaps in future

Reply:
might do as part of the collections work, since collections will need similar treatment. we can make it nice and parallel